### ISSUE TITLE
resource azurerm_kubernetes_cluster do not get kube_config from AAD enabled v1.24 cluster

### ISSUE THREAD

**Issue by GiuseppeChiesa-TomTom** [hashicorp/terraform-provider-azurerm#18139]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.4

### AzureRM Provider Version

3.20.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {
    resource_group {
      prevent_deletion_if_contains_resources = false
    }
  }
}


resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_kubernetes_cluster" "example" {
  name                = "example-aks1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "exampleaks1"
  kubernetes_version  = "1.24.3"

  azure_active_directory_role_based_access_control {
    azure_rbac_enabled = true
    managed            = true
  }

  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_D2_v2"
  }

  identity {
    type = "SystemAssigned"
  }

}

resource "null_resource" "test" {
  provisioner "local-exec" {
    command = "echo ${azurerm_kubernetes_cluster.example.kube_config.0.host}"
  }
  triggers = {
    always = timestamp()
  }
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: Invalid index
│ 
│   on main.tf line 70, in resource "null_resource" "test":
│   70:     command = "echo ${azurerm_kubernetes_cluster.example.kube_config.0.host}"
│     ├────────────────
│     │ azurerm_kubernetes_cluster.example.kube_config has a sensitive value
│ 
│ The given key does not identify an element in this collection value.
╵
```

### Expected Behaviour

Get kube_config properly populated and saved in the state file

### Actual Behaviour

I checked kube_config in tf state. It was empty.

```
"kube_config": []
```

I think it's related to https://github.com/hashicorp/terraform-provider-azurerm/issues/17182 and it can be reproduced with the K8S 1.24.3 

The issue seems to be solved but the solution is limited to the datasource. Here the problem is to the output of the azurerm_kubernetes_cluster resource.

### References

- https://github.com/hashicorp/terraform-provider-azurerm/issues/17182
- specific comment of another user reporting possibly the same issue: https://github.com/hashicorp/terraform-provider-azurerm/issues/17182#issuecomment-1227889168

**Comment by DaleyKD** [comment]
Will this handle the `exec` stuff?

Correct me if I'm wrong, but with AKS 1.24.x, we are being forced to get a token from `kubelogin`, meaning we have to exec `kubelogin`. The `kube_config_raw` comes back nicely and that's great. We can put that in a `.kubeconfig` file or whatever and then future modules (and providers like `kubernetes` or `helm`) can use that file (I realize it's not best practice).

But if we parse `kube_config_raw` into a `kube_config` block, if it doesn't have the `exec` block, can we use this in the same module like we have been doing?

(Am I making sense? I'm just rambling here.)

**Comment by DaleyKD** [comment]
@mbfrahry : Would you mind reading my comment and answering the questions? TIA.

**Comment by mbfrahry** [comment]
Hi @DaleyKD, Kubernetes is not something I'm at all an expert on so I can't answer your question. I'd suggest trying to replicate the situation you're describing to see what happens. I'm sorry I can't be more helpful

**Comment by GiuseppeChiesa-TomTom** [comment]
Any idea how this change will be released? I see is targeting 3.21 milestone, but it should be actually part of bug fixes.
